src/Nerdbank.MessagePack/StandardVisitor.cs

```diff
@@ -80,8 +80,23 @@ internal StandardVisitor(ConverterCache owner, TypeGenerationContext context)
 
 		IConstructorShape? ctorShape = objectShape.Constructor;
 
-		Dictionary<string, IConstructorParameterShape>? ctorParametersByName =
-			ctorShape?.Parameters.ToDictionary(p => p.Name, StringComparer.OrdinalIgnoreCase);
+		Dictionary<string, IConstructorParameterShape>? ctorParametersByName = null;
+		if (ctorShape is not null)
+		{
+			ctorParametersByName = new(StringComparer.OrdinalIgnoreCase);
```

Comment:
Is hardcoding case insensitive matching desirable in the general case?

Reply:
Good question. I tried changing this to `Ordinal` (and one other place) and all my tests still pass. So I guess I'll switch to `Ordinal` since that seems safer, until I have tests that prove that's undesirable. Thanks for the suggestion.